tensorflow/compiler/xla/service/gpu/instruction_fusion.cc

```diff
@@ -29,12 +29,23 @@ limitations under the License.
 namespace xla {
 namespace gpu {
 
+bool ElementIsF32OrF16(const Shape& shape) {
+  PrimitiveType type = shape.element_type();
+  return type == F32 || type == F16;
+}
+
 /*static*/ bool GpuInstructionFusion::IsExpensive(
     const HloInstruction& instruction) {
-  // We say that floating-point division is cheap on the GPU.
-  if (instruction.opcode() == HloOpcode::kDivide &&
-      ShapeUtil::ElementIsFloating(instruction.shape())) {
-    return false;
+
+  // We say that some floating-point math ops are cheap on the GPU.
+  switch (instruction.opcode()) {
```

Comment:
Please add the rationale you mentioned in the PR (that these lower to single instructions).

Reply:
Make sense. Will add.